src/common/users.ts

```diff
@@ -400,15 +400,14 @@ export const checkAndClearUserStreak = async (
   info: GraphQLResolveInfo,
   streak: GQLUserStreakTz,
 ): Promise<boolean> => {
-  const { lastViewAtTz: lastViewAt, timezone } = streak;
+  const { lastViewAtTz: lastViewAt, currentDateTz: currentDate } = streak;
 
   if (!lastViewAt) {
     return false;
   }
 
-  const today = getTodayTz(timezone);
-  const day = today.getDay();
-  const difference = differenceInDays(today, lastViewAt);
+  const day = currentDate.getDay();
+  const difference = differenceInDays(currentDate, lastViewAt);
```

Comment:
Tests fail because since you know return date from psql `useFakeTimers` no longer works so you get wrong dates.

Reading slack and this PR I don't think the issue is computing in sql or JS. Basically in both cases you want to get midnight in user timezone?

Reply:
yeah, I was trying to fix specs, but this is not trivial at all if we get the current time from SQL, which is a pain. 

The reason for this change is due to this part here: https://github.com/dailydotdev/daily-api/blob/264132ab3261b3c78f34a95fa6e64ab9ca28c936/src/common/date.ts#L4 

I am not sure that when we create a new date object based on the one converted to the user timezone, we are not converting it back to UTC again, which then causes issues when we do comparisons.

Reply:
Yeah, dates from database are tricky to mock, I had this issue myself when doing personalized digest. If your onlyy concern is the timezone preservation you can replace `getTodayTz` with following implementation:
```ts
export const getTodayTz = (timeZone: string) => {
  const now = new Date();
  const timeZonedToday = utcToZonedTime(now, timeZone);
  timeZonedToday.setHours(0, 0, 0, 0);

  return timeZonedToday;
};
```

This makes it more clear and I am sure the timezone is preserved.